### PR TITLE
FEAT: Add Search Feature to the product listing page

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

A search is used to search for a specific product

### 🚀 New behavior

- The user could search for products from product listing page
- Improved the folder structure for Utilities and Services Folders

### 🖼 Preview Link

https://viskart-react-version-3cyj8kuds-vishalpatil18.vercel.app/

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue
- #22 

> NOTE: No need to review the services and utility files as just their folders are changed

### CODE REVIEW

**Review by Abhijeet43 — APPROVED** [review]
- The functionality is working as intended!!
- Naming Conventions are good
- Directory Structure and Code are clean

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
 import { AuthModal } from "../../components";
 import { ASSETS_URL, ICONS_URL } from "../../constants";
-import { useAuth, useAuthModal } from "../../context";
+import { useAuth, useAuthModal, useFilter } from "../../context";
 import { toast } from "react-toastify";
-import { Link } from "react-router-dom";
+import { Link, useLocation, useNavigate } from "react-router-dom";
```
Comment (kart-c) [inline src/components/Navbar/Navbar.jsx:5]: `useNavigate` is not used in this file. If it's an unused import, you can delete it.

**Review by kart-c — APPROVED** [review]
The search functionality is really amazing. Also, I just noticed that, since you are using `text ellipsis` for the card name, you have provided the `title` attribute to let the user know what the full name of the product is.  This is a very good UX point

File: src/components/Navbar/Navbar.jsx (hunk 1 above)
Comment (VishalPatil18) [reply src/components/Navbar/Navbar.jsx:5]: Thanks for pointing it out! I've removed that now! I will push that in the next PR.

**Pull request merged**